private/ft_fetch_sens.m

```diff
@@ -113,66 +132,33 @@
 elseif hasdatagrad
   display('using gradiometers specified in the data\n');
   sens = data.grad;
+  
 elseif haselecfile
   display('reading electrodes from file ''%s''\n', cfg.elecfile);
   sens = ft_read_sens(cfg.elecfile);
   % only keep positions and labels in case of EEG electrodes
-  dum  = sens;
-  sens = [];
-  if isfield(dum,'chanpos')
-  sens.chanpos = dum.chanpos;
-  end
-  if isfield(dum,'elecpos')
-  sens.elecpos = dum.elecpos;
-  end
-  sens.label   = dum.label;
-  if isfield(dum,'unit')
-    sens.unit = dum.unit;
-  end
-  if isfield(dum,'coordsys')
-    sens.coordsys = dum.coordsys;
-  end
+  sens = keepfields(sens, {'elecpos', 'chanpos', 'unit', 'coordsys', 'label'});
 elseif hascfgelec
   display('using electrodes specified in the configuration\n');
   sens = cfg.elec;
   % only keep positions and labels in case of EEG electrodes
-  dum  = sens;
-  sens = [];
-  if isfield(dum,'chanpos')
-  sens.chanpos = dum.chanpos;
-  end
-  if isfield(dum,'elecpos')
-  sens.elecpos = dum.elecpos;
-  end
-  sens.label   = dum.label;
-  if haselectra
-    sens.tra = dum.tra;
-  end
-  if isfield(dum,'unit')
-    sens.unit = dum.unit;
-  end
-  if isfield(dum,'coordsys')
-    sens.coordsys = dum.coordsys;
-  end
+  sens = keepfields(sens, {'elecpos', 'chanpos', 'unit', 'coordsys', 'label'});
 elseif hasdataelec
   display('using electrodes specified in the data\n');
   sens = data.elec;
   % only keep positions and labels in case of EEG electrodes
-  dum  = sens;
-  sens = [];
-  if isfield(dum,'chanpos')
-  sens.chanpos = dum.chanpos;
-  end
-  if isfield(dum,'elecpos')
-  sens.elecpos = dum.elecpos;
-  end
-  sens.label   = dum.label;
-  if isfield(dum,'unit')
-    sens.unit = dum.unit;
-  end
-  if isfield(dum,'coordsys')
-    sens.coordsys = dum.coordsys;
-  end
+  sens = keepfields(sens, {'elecpos', 'chanpos', 'unit', 'coordsys', 'label'});
+  
+elseif hasoptofile
+  display('reading optodes from file ''%s''\n', cfg.optofile);
+  sens = ft_read_sens(cfg.elecfile);
+elseif hascfgopto
+  display('using optodes specified in the configuration\n');
+  sens = cfg.opto;
```

Comment:
should this be added? 
sens = keepfields(sens, {'optopos', 'optotype', 'chanpos', 'unit', 'coordsys', 'label', 'transceiver', 'wavelength', 'transmits', 'laserstrength'});